front/lib/plans/usage/mau.ts

```diff
@@ -0,0 +1,90 @@
+import type { LightWorkspaceType, Result } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+import { QueryTypes } from "sequelize";
+import type Stripe from "stripe";
+
+import { updateStripeActiveUsersForSubscriptionItem } from "@app/lib/plans/stripe";
+import type { MauReportUsageType } from "@app/lib/plans/usage/types";
+import { InvalidRecurringPriceError } from "@app/lib/plans/usage/types";
+import { getFrontReplicaDbConnection } from "@app/lib/resources/storage";
+
+export async function getActiveUsersInWorkspaceForPeriod({
+  messagesPerMonthForMau,
+  since,
+  to,
+  workspace,
+}: {
+  messagesPerMonthForMau: number;
+  since: Date;
+  to?: Date;
+  workspace: LightWorkspaceType;
+}): Promise<number> {
+  // Use the replica database to compute this avoid impacting production performances.
+  const result = await getFrontReplicaDbConnection().query(
```

Comment:
I know we don't like sequelize and we intend to move from it, but I feel a bit uneasy mixing raw sql queries and Sequelize model-based queries. IMHO it is preferable to consistently use sequelize everywhere, then resources everywhere when they become available, then drizzle everywhere when the time comes

Reply:
Also here I think it would be a bit easier to read / understand the full logic (which I agree is not always true)

Reply:
In this particular scenario, I question whether using the high-level Sequelize API is better. Given the complexity of the query, using Sequelize will probably complicate it and make the code difficult to read.

I agree with the fact that mixing raw SQL queries and Sequelize model-based queries can feel somewhat uneasy, with a few exceptions. This case might be one of those exceptions where using a raw SQL query could be more appropriate.